python/hsfs/core/statistics_engine.py

```diff
@@ -72,38 +72,51 @@ def compute_statistics(
                 stats = statistics.Statistics(
                     commit_time=commit_time,
                     content=content_str,
-                    feature_group_commit_id=feature_group_commit_id,
+                    window_end_commit_id=feature_group_commit_id,
                 )
-                self._save_statistics(stats, metadata_instance, feature_view_obj)
+                return self._save_statistics(stats, metadata_instance, feature_view_obj)
         else:
             # Python engine
             engine.get_instance().profile_by_spark(metadata_instance)
 
     def compute_single_feature_statistics(
         self,
+        metadata_instance,
         feature_dataframe,
         feature_name,
-    ) -> FeatureDescriptiveStatistics:
-        """Compute statistics for a single feature without sending the result to Hopsworks.
+        start_time,
+        end_time,
+        row_percentage,
+    ) -> statistics.Statistics:
+        """Compute statistics for a single feature and sends the result to Hopsworks.
 
         Args:
             feature_dataframe: Single-feature Spark or Pandas DataFrame to compute the statistics on.
             feature_name: str. Name of the feature.
+            start_time: int: Window start commit time
+            end_time: int: Window end commit time
+            row_percentage: Percentage of rows to include
```

Comment:
for percentage can we add that the range is 0 to 1 in decimal and not o to 100

Reply:
Done